lib/providers/sensors_provider.dart

```diff
@@ -0,0 +1,269 @@
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:sensors_plus/sensors_plus.dart';
+
+class SensorsProvider extends ChangeNotifier{
+  late List<_PositionData> _currentPosition;
+  late List<_AccelerometerData> _accelerometerEvents;
+  late List<_UserAccelerometerData> _userAccelerometerEvents;
+  late List<_MagnetometerData> _magnetometerEvents;
+  late List<_GyroscopeData> _gyroscopeEvents;
+
+  static const Duration _ignoreDuration = Duration(milliseconds: 20);
+  
+  bool _run = false;
+
+  DateTime? _userAccelerometerUpdateTime;
+  DateTime? _accelerometerUpdateTime;
+  DateTime? _gyroscopeUpdateTime;
+  DateTime? _magnetometerUpdateTime;
+  DateTime? _currentPositionUpdateTime;
+
+  int? _userAccelerometerLastInterval;
+  int? _accelerometerLastInterval;
+  int? _gyroscopeLastInterval;
+  int? _magnetometerLastInterval;
+  final _streamSubscriptions = <StreamSubscription<dynamic>>[];
+
+  Duration sensorInterval = SensorInterval.normalInterval;
+  void _initState() {
+    _streamSubscriptions.add(
+      userAccelerometerEventStream(samplingPeriod: sensorInterval).listen(
+            (UserAccelerometerEvent event) {
+          final now = DateTime.now();
+          _UserAccelerometerData userAccelerometerData = _UserAccelerometerData(event, now);
+          _userAccelerometerEvents.add(userAccelerometerData); // Add event to the list
+          if (_userAccelerometerUpdateTime != null) {
+            final interval = now.difference(_userAccelerometerUpdateTime!);
+            if (interval > _ignoreDuration) {
+              _userAccelerometerLastInterval = interval.inMilliseconds;
+            }
+          }
+          _userAccelerometerUpdateTime = now;
+          //notifyListeners(); // Notify listeners of state change
```

Comment:
why this commented?

Reply:
We use this provider only to separate view and logic. The changes in the provider do not affects the state of the widget.

Reply:
this is very true.
only notify listeners if it will affect the UI widgets